src/commands/gate/evaluate.ts

```diff
@@ -119,22 +125,25 @@ export class GateEvaluateCommand extends Command {
   }
 
   private async evaluateRules(api: APIHelper, evaluateRequest: Payload): Promise<number> {
-    if (this.shouldWait()) {
-      this.context.stdout.write(renderWaiting())
-      await this.delay(this.waitingTime)
-    }
-
     this.context.stdout.write(renderGateEvaluationInput(evaluateRequest))
 
-    return retryRequest(
-      () => api.evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout)),
-      {
-        onRetry: (e, attempt) => {
+    /**
+     * `retryRequest` does not allow setting a wait time dependent on a backend response,
+     * so we handle the wait time in `evaluateRulesWithWait`: we'll wait whatever is necessary 
+     * for the returned promise to resolve or be rejected. The retry will start immediately after
```

Comment:
my bad: there's only a retry if the promise is rejected
```suggestion
     * for the returned promise to be rejected. The retry will start immediately after
```